.github/workflows/lint_python.yml

```diff
@@ -7,4 +7,6 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v1
       - run: pip install flake8
-      - run: flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+      - run: |
+          EXCLUDE=./.*,./atest/testdata/*
+          flake8 . --count --exclude=$EXCLUDE --select=E9,F63,F7,F82 --show-source --statistics
```

Comment:
Could this just be run for `src/robot` instead of of trying to exclude other directories?

Reply:
Do we just ignore problems that appear in `setup.py` for instance?  I think it is safer to only exclude certain test directories where you are consciously trying to break things.  Also, should we just paper over the fact that this code does not follow the Python porting best practice [___use feature detection instead of version detection___](https://docs.python.org/3/howto/pyporting.html#use-feature-detection-instead-of-version-detection)?

Reply:
1. If you think it's important to test `setup.py`, run linter for it in addition to `src/robot`. I want linting to be opt-in, not opt-out, in general and I definitely don't want it for *any* code under atest or utest directories.

2. I'm personally even sure does it make sense to lint `setup.py`. Do you have examples of problems that linter could find that wouldn't be found otherwise? Now it only found a non-bug and an annoying magic comment is needed to silence it. 

3. This code base uses both version detection and feature detection, depending on which is more practical. Most of the Python 2/3 differences are such that version detection is 100% safe, results with cleaner code, and is also a bit faster (`try/except` isn't free). Our `PY3` constant is defined so that it will be true also with possible Python 4, so as long it doesn't break compatibility with Python 3 we are safe. And if it breaks, we need to fix our code anyway. Finally, this stuff is definitely not going to change before RF 4.0 when we remove Python 2 support altogether.